sklearn/preprocessing/_data.py

```diff
@@ -2350,9 +2352,13 @@ def transform(self, K, copy=True):
         """
         check_is_fitted(self)
 
-        K = self._validate_data(K, copy=copy, dtype=FLOAT_DTYPES, reset=False)
+        xp, _ = get_namespace(K)
+
+        K = self._validate_data(
+            K, copy=copy, dtype=_array_api.supported_float_dtypes(xp), reset=False
+        )
 
-        K_pred_cols = (np.sum(K, axis=1) / self.K_fit_rows_.shape[0])[:, np.newaxis]
+        K_pred_cols = (xp.sum(K, axis=1) / self.K_fit_rows_.shape[0])[:, None]
```

Comment:
https://data-apis.org/array-api/latest/API_specification/generated/array_api.constants.newaxis.html?highlight=newaxis

can be replaced with xp.newaxis for readability

Reply:
I actually tried with `xp.newaxis` initially, but I got an `AttributeError` both with `torch` and `numpy`

```
AttributeError: module 'array_api_compat.torch' has no attribute 'newaxis'
AttributeError: module 'numpy.array_api' has no attribute 'newaxis'
```

@betatim do you know why `newaxis` may not be included?

Reply:
Unfortunately I don't know why `newaxis` doesn't exist for them. I would have guessed "bug in array_api_compat" if that was the only place where its missing, but I think of`numpy.array_api` as the "very much exactly what the standard says" implementation. One option is that it is something that only got added in a newer version of the standard

Reply:
I opened https://github.com/data-apis/array-api-compat/issues/64 upstream.

Ok for using `None` in scikit-learn in the mean time.